src/3d/terrain/qgsflatterraingenerator.cpp

```diff
@@ -133,6 +146,39 @@ void QgsFlatTerrainGenerator::readXml( const QDomElement &elem )
   // crs is not read/written - it should be the same as destination crs of the map
 }
 
+QVector<QgsChunkNode *> QgsFlatTerrainGenerator::createChildren( QgsChunkNode *node ) const
+{
+  QVector<QgsChunkNode *> children;
+
+  if ( node->level() >= mMaxLevel )
+    return children;
+
+  const QgsChunkNodeId nodeId = node->tileId();
+  const float childError = node->error() / 2;
+  const QgsAABB bbox = node->bbox();
+  float xc = bbox.xCenter(), zc = bbox.zCenter();
+
+  for ( int i = 0; i < 4; ++i )
+  {
+    int dx = i & 1, dy = !!( i & 2 );
+    const QgsChunkNodeId childId( nodeId.d + 1, nodeId.x * 2 + dx, nodeId.y * 2 + ( dy ? 0 : 1 ) );  // TODO: inverse dy?
```

Comment:
This code was pulled from the parent class, `QgsQuadtreeChunkLoaderFactory::createChildren()` and added the mExtent based filter at the end, so the todo is some old leftover :)
Maybe I could investigate moving the mExtent to the parent class and apply the filtering directly there instead of overriding it.